File: sws_backend/SWS/src/main/java/it/unisa/emad/comunesalerno/sws/dto/ServizioMappaDTO.java
package it.unisa.emad.comunesalerno.sws.dto;

import it.unisa.emad.comunesalerno.sws.entity.Area;
import it.unisa.emad.comunesalerno.sws.entity.Servizio;
import lombok.*;

import java.util.List;

@Data
@NoArgsConstructor

public class ServizioMappaDTO {
    private String id;
    private String nome;
    private String struttura;
    private String ente;

    private String posizione;
    private String indirizzo;


    private String customIcon;
    public ServizioMappaDTO(String id, String nome, String struttura, String ente,  String indirizzo,String posizione,  String customIcon) {
        this.id = id;
        this.nome = nome;
        this.struttura = struttura;
        this.ente = ente;
        this.posizione = posizione;
        this.indirizzo = indirizzo;
        this.customIcon=customIcon;
    }
}

File: sws_frontend/lib/services/dto/ServizioMappaDTO.dart
// To parse this JSON data, do
//
//     final servizioMappaDto = servizioMappaDtoFromJson(jsonString);

import 'dart:convert';

import 'package:flutter/cupertino.dart';

import '../entity/Area.dart';

ServizioMappaDto servizioMappaDtoFromJson(String str) => ServizioMappaDto.fromJson(json.decode(str));

String servizioMappaDtoToJson(ServizioMappaDto data) => json.encode(data.toJson());

class ServizioMappaDto {
  ServizioMappaDto({
    required this.id,
    required this.nome,
    required this.struttura,
    required this.indirizzo,
    required this.ente,
    this.posizione,
    this.customIcon
  });

  String id;
  String nome;
  String struttura;
  String ente;
  String indirizzo;
  String? posizione;
  String? customIcon;

  factory ServizioMappaDto.fromJson(Map<String, dynamic> json) => ServizioMappaDto(
    id: json["id"],
    nome: json["nome"],
    struttura: json["struttura"],
    ente: json["ente"],
    posizione: json["posizione"],
    customIcon: json["customIcon"],
    indirizzo: json["indirizzo"],
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "nome": nome,
    "struttura": struttura,
    "ente": ente,
    "posizione": posizione,
    "indirizzo": indirizzo,
    "customIcon": customIcon,
  };
  IconData? getIconData(){

    return customIcon!=null?IconData(int.parse(customIcon!),fontFamily: "MaterialIcons"):null;
  }
}
